### PR TITLE
Undo button visibility

### LINKED ISSUE #1480: undo stacking during startup
Step 1: solve two different quest

Step 2: Switch applications, use system killer of apps

Step 3: Start StreetComplete, as it starts keep clicking "Undo" button

As result multiple undo windows will appear, each allowing to keep undoing. It is possible to undo more times than solved quests and get negative star count.

### PR DESCRIPTION
**Pull request by westnordost**
Fixes #1480, #1433

### CODE REVIEW

File: app/src/main/java/de/westnordost/streetcomplete/MainActivity.java [hunk 1]
```diff
@@ -849,6 +869,7 @@ private void showMarkerSolvedAnimation(@DrawableRes int iconResId, PointF startS
 		flingQuestMarkerTo(img, answersCounter.getAnswerTarget(), () -> {
 			root.removeView(img);
 			answersCounter.addOneUnsynced(source);
+			btnUndo.setVisible(true);
```
Comment (westnordost) [inline app/src/main/java/de/westnordost/streetcomplete/MainActivity.java]: Not all quests can be undone. I.e. note quests can't. Maybe need to call `updateUndoButtonVisibility();` instead, or find out first if the solved quest is something undoable.

**Comment by matkoniecz** [comment]
One problem that I spotted is that in case where the first solved quest is a note one undo button still appears.

**Comment by westnordost** [comment]
Yes exactly, but it is fixed by my last commit, is it not?

**Comment by matkoniecz** [comment]
I retested it now and it still happens. Tested by uninstalling dev version of SC, installing dev version compiled from 184005a270f65ec056934a710de8844fe019e38b selecting single note quest, typing gibberish and selecting OK.


Curiously, switching apps and killing SC then entering it is NOT showing undo button but showing one more note quest is showing it again.

Solving normal quest, switching apps and killing SC then entering it is showing undo button.

**Comment by westnordost** [comment]
Ok I will have a look at it tomorrow

**Pull request merged**